File: updater.php
<?php
	//将下面的信息修改为DNS-O-Matic的用户信息
	$username = "123";
	$password = "123";
	$hostname[] = "a.example.com";		//在dnsomatic添加服务，然后在这里填写域名
	$hostname[] = "b.example.net";
	$hostname[] = "c.example.org";

	$ip = get_publicIP();

	if($ip == "0")
	{
		echo "Error!Cann't get public IP!\n";
	}else
	{
		for($i=0;$i<count($hostname);$i++)
		{
			$dnsomaticapi = "https://$username:$password@updates.dnsomatic.com/nic/update?hostname=$hostname[$i]&myip=$ip&wildcard=NOCHG&mx=NOCHG&backmx=NOCHG";
			echo $hostname[$i]." ".http_request($dnsomaticapi)."\n";
		}
	}

	function http_request($url, $data = null)
	{
	    $curl = curl_init();
	    curl_setopt($curl, CURLOPT_URL, $url);
	    curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, FALSE);
	    curl_setopt($curl, CURLOPT_SSL_VERIFYHOST, FALSE);
	    if (!empty($data)){
	        curl_setopt($curl, CURLOPT_POST, 1);
	        curl_setopt($curl, CURLOPT_POSTFIELDS, $data);
	    }
	    curl_setopt($curl, CURLOPT_RETURNTRANSFER, TRUE);
	    $output = curl_exec($curl);
	    curl_close($curl);
	    return $output;
	}

	function get_publicIP()
	{
		$json = file_get_contents("http://api.k780.com/?app=ip.local&appkey=10003&sign=b59bc3ef6191eb9f747dd4e83c99f2a4&format=json");
		$info = json_decode($json,true);
		if($info["success"]=="1")
		{
			$result = $info["result"];
			return $result["ip"];
		}else
		{
			$json = file_get_contents("http://ip.taobao.com/service/getIpInfo.php?ip=myip");
			$info = json_decode($json,true);
			if($info["code"]=="0")
			{
				$data = $info["data"];
				return $data["ip"];
			}else{
				return "0";
			}
		}
	}
?>

File: updater.py
# -*- coding: utf-8 -*-
import urllib.request
import json

def get_publicIP() :
	jsondata = urllib.request.urlopen('http://api.k780.com/?app=ip.local&appkey=10003&sign=b59bc3ef6191eb9f747dd4e83c99f2a4&format=json').read().decode('utf-8')
	info = json.loads(jsondata)
	if info['success'] :
		result = info['result']
		return result['ip']
	else :
		jsondata = urllib.request.urlopen('http://ip.taobao.com/service/getIpInfo.php?ip=myip').read().decode('utf-8')
		info = json.loads(jsondata)
		if info['code'] :
			result = info['data']
			return result['ip']
		else :
			return "0"


#将下面的信息修改为DNS-O-Matic的用户信息
username = '123'
password = '123'
hostname = ['a.example.com',
			'b.example.com',
			'c.example.com']

ip = get_publicIP()
apiurl = 'https://updates.dnsomatic.com'
password_mgr = urllib.request.HTTPPasswordMgrWithDefaultRealm()
password_mgr.add_password(None, apiurl, username, password)
handler = urllib.request.HTTPBasicAuthHandler(password_mgr)
opener = urllib.request.build_opener(handler)

if ip=="0" :
	print("Error!Cann't get public IP!")
else :
	for host in hostname:
		url = "https://updates.dnsomatic.com/nic/update?hostname="+host+"&myip="+ip+"&wildcard=NOCHG&mx=NOCHG&backmx=NOCHG"
		dnsomaticapi = opener.open(url).read().decode('utf-8')
		print(host+" "+dnsomaticapi)
		
